### ISSUE TITLE
Grab project logs from start of file and not end with the fluentd collector

### ISSUE THREAD

**Issue by rayisbadat** [rancher/rancher#25048]
**What kind of request is this (question/bug/enhancement/feature request):**
Feature Request/Enhancement

**Steps to reproduce (least amount of steps as possible):**
1) Setup Project Logging via Rancher to remote collector.  Doesn't matter type.
2) Set flush interval to 1 second.
2) Deploy a workflow that generates lots of output ( I used my test image rayisbadat/infloop )
  
**Result:**
There is a delay before the fluentd log collector starts forwarding logs.  Delay ranges from 10-60 seconds, although had some outliers that went two minutes.   This causes us to lose the boot output, which we think will be critical to roll out and production.

**Other details that may be helpful:**
Adding two settings into auto generated `/opt/rke/var/lib/kubelet/pods/UUID/volumes/kubernetes.io~secret/config/project.conf`  will fix this problem in the auto generated  fluentd project.conf file.  

 `skip_refresh_on_startup true` Speeds up the collection and forwarding
 `read_from_head true` Causes fluentd collector to scrape entire file and not start with the end of file.  

PR: https://github.com/rancher/rancher/pull/25049

Timing from some simple tests (time in seconds)

|setting\Run # |1|2|3|4|5|min|avg|mean|max
|--------------|-|-|-|-|-|---|---|----|---
|default|58.203|30.385|23.806|37.457|13.32|13.32|32.6342|30.385|58.203
|skip_refresh_on_startup true|48.933|39.976|10.573|43.578|38.641|10.573|36.3402|39.976|48.933
|read_from_head true|55.677|58.089|56.444|12.87|30.588|12.87|42.7336|55.677|58.089
|Both|15.166|19.32|26.926|33.81|34.274|15.166|25.8992|26.926|34.274

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):v2.3.4
- Installation option (single install/HA): Single install

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom(Self Hoststed)
- Machine type (cloud/VM/metal) and specifications (CPU/memory): VM (Virtual Box), 2 Cores, 4GB of Ram
- Kubernetes version (use `kubectl version`): v1.17.0

- Docker version (use `docker version`):
```
Client: Docker Engine - Community
 Version:           19.03.4
 API version:       1.40
 Go version:        go1.12.10
 Git commit:        9013bf583a
 Built:             Fri Oct 18 15:49:05 2019
 OS/Arch:           linux/amd64
 Experimental:      false
```

**Comment by loganhz** [comment]
Similar issue https://github.com/rancher/rancher/issues/24318

**Comment by izaac** [comment]
Rancher version: `2.4 master-head (02/28/2020)` commit id: e59944099

- tailed the external fluentd docker container aggregator logs to monitor the logs.
- Enabled project logging in the Rancher `System` project with an external fluentd aggregator.
- Checked a couple `System` project containers logs to verify the start of file logs
- Exported the fluentd docker container aggregator logs to a local file
- Compared the logs in the fluentd aggregator local file with the `System` project containers logs

Now the logs are getting sent and aggregated by the external fluentd from the beginning of the logs in the rancher containers.

**Comment by izaac** [comment]
Further testing was done today in Rancher version: `2.4 master-head (03/02/2020)` commit id: fa96c0c3d

- Setup the external Fluentd aggregator to use `@type file`  to redirect the logs to a log file
- Deploy the provided `rayisbadat/infloop` image for generating logs.
- Setup the project level logging to start collecting the logs and send to the external Fluentd.
- Monitored the logs size until I got more than 50 Mb in file size.
- Deleted the infloop workload from rancher
- Started analizing the logfile.


I was able to see the start of the logs of the infloop image `{"log":"Output> Iteration: 1...` even after it got very big in size. Validating that the logs is collected from start of file and not tailing the log.

**Comment by izaac** [comment]
I was also able to reproduce this issue in Rancher v2.3.5  instead of starting on log iteration 1 as my [previous test](https://github.com/rancher/rancher/issues/25048#issuecomment-593505436) in 2.4 we only got the part of the logs in the aggregator the first one being: `{"log":"Output> Iteration: 73777...`

**Comment by izaac** [comment]
Upgrading from Rancher v2.3.5 to latest `2.4 master-head (03/03/2020)` commit id: c921ba250 I see the new configuration getting applied and working as expected.

**Comment by deniseschannon** [comment]
@izaac can you test this in 2.3-hea

**Comment by izaac** [comment]
Rancher version: `v2.3-head (03/13/2020)` commit id: cdd0acc06

The issue was previously reproduced in 2.3.5.

I performed the following test scenarios with positive results:

1. Deploy the provided `rayisbadat/infloop` image for generating logs.
2. Setup the external Fluentd aggregator to use `@type file` to redirect the logs to a log file
3. Setup the project level logging to start collecting the logs and send to the external Fluentd.
4. Monitored the logs size until I got more than 50 Mb in file size.
5. Deleted the infloop workload from rancher
6. Started analyzing the log file.

Upgrade was made from v2.2.9 to v2.3-head I see the new configuration getting applied and working as expected.
